src/components/SubStepForms/RegistrationNumberStep.tsx

```diff
@@ -0,0 +1,133 @@
+import {useFocusEffect} from '@react-navigation/native';
+import React, {useCallback, useRef} from 'react';
+import {View} from 'react-native';
+import FormProvider from '@components/Form/FormProvider';
+import InputWrapper from '@components/Form/InputWrapper';
+import type {FormInputErrors, FormOnyxKeys, FormOnyxValues} from '@components/Form/types';
+import Icon from '@components/Icon';
+import * as Expensicons from '@components/Icon/Expensicons';
+import type {AnimatedTextInputRef} from '@components/RNTextInput';
+import Text from '@components/Text';
+import TextInput from '@components/TextInput';
+import TextLink from '@components/TextLink';
+import useLocalize from '@hooks/useLocalize';
+import type {SubStepProps} from '@hooks/useSubStep/types';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {getFieldRequiredErrors, isValidRegistrationNumber} from '@libs/ValidationUtils';
+import CONST from '@src/CONST';
+import type {Country} from '@src/CONST';
+import type {OnyxFormValuesMapping} from '@src/ONYXKEYS';
+
+type RegistrationNumberStepProps<TFormID extends keyof OnyxFormValuesMapping> = SubStepProps & {
+    /** The ID of the form */
+    formID: TFormID;
+
+    /** A function to call when the form is submitted */
+    onSubmit: (values: FormOnyxValues<TFormID>) => void;
+
+    /** The ID of the form input */
+    inputID: FormOnyxKeys<TFormID>;
+
+    /** The default values for the input */
+    defaultValue: string;
+
+    /** Country of affiliated policy */
+    country: Country | '';
+
+    /** Whether to delay autoFocus to avoid conflicts with navigation animations */
+    shouldDelayAutoFocus?: boolean;
+};
+
+function RegistrationNumberStep<TFormID extends keyof OnyxFormValuesMapping>({
+    formID,
+    onSubmit,
+    inputID,
+    defaultValue,
+    isEditing,
+    country,
+    shouldDelayAutoFocus = false,
+}: RegistrationNumberStepProps<TFormID>) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const theme = useTheme();
+
+    const internalInputRef = useRef<AnimatedTextInputRef>(null);
+    const focusTimeoutRef = useRef<NodeJS.Timeout | null>(null);
+
+    useFocusEffect(
+        useCallback(() => {
+            if (!shouldDelayAutoFocus) {
+                return;
+            }
+
+            focusTimeoutRef.current = setTimeout(() => {
+                internalInputRef.current?.focus();
+            }, CONST.ANIMATED_TRANSITION);
+
+            return () => {
+                if (!focusTimeoutRef.current) {
+                    return;
+                }
+                clearTimeout(focusTimeoutRef.current);
+            };
+        }, [shouldDelayAutoFocus]),
+    );
```

Comment:
Could we reuse this logic? Probably create a hook, I saw you used it on another places as well